src/conftest.py

```diff
@@ -17,6 +19,38 @@
 from test.conftest import merge_markers
 
 
+@pytest.fixture(scope='function', autouse=True)
+def _assert_no_leftover_processes():
+    # FIXME: Our tests violate any assertions in this function.
+    #        Thus they are commented out.
+    current_process = psutil.Process()
+
+    # From e.g. class scoped fixtures ther might be some processes already started
+    children_before = current_process.children(recursive=True)
+    pids_before = {process.pid for process in children_before}
+
+    yield
+
+    children_after = current_process.children(recursive=True)
+    pids_after = {process.pid for process in children_after}
+
+    # assert len(pids_before - pids_after) == 0, "Tests should not kill subprocesses that they didn't start"
+
+    leftover_pids = pids_after - pids_before
+    leftover_processes = [psutil.Process(pid) for pid in leftover_pids]
+    # Give them some time to shutdown
+    if len(leftover_processes) != 0:
+        # XXX Does this work?
+        # Shtudown out of scope managers etc.
+        gc.collect()
+        time.sleep(2)
+
+    # for process in leftover_processes:
+    #     process.kill()
+
+    # assert len(leftover_processes) == 0, 'Leftover processes'
```

Comment:
> Do you plan to keep it in the code?

Idc. It was in here since I guessed the zombies somehow caused the CI errors.
This code is also pasted in #926 so I'd say lets drop it from this PR.

Reply:
Removed.